main.go

```diff
@@ -266,15 +269,14 @@ func (ctx *httpContext) OnHttpResponseBody(bodySize int, endOfStream bool) types
 		}
 	}
 
-	// Response  body has to be buffered in order to check that it is fully legit
+	// Response body has to be buffered in order to check that it is fully legit
 	if !endOfStream {
-		// TODO(M4tteoP): Address response body interruption logic after https://github.com/corazawaf/coraza-proxy-wasm/issues/26
-		// return types.ActionPause
-		return types.ActionContinue
+		// TODO(M4tteoP): Update response body interruption logic after https://github.com/corazawaf/coraza-proxy-wasm/issues/26
+		return types.ActionPause
```

Comment:
I guess we should change request body to also have the same behavior

https://github.com/corazawaf/coraza-proxy-wasm/blob/main/main.go#L190

Reply:
Yes, definitely. Just like the response body, it is needed in order to send upstream only after the coraza feedback.
That being said, my concern is about buffering twice the bodies (envoy and coraza side). Next step would maybe be to buffer it only once (coraza side) and populate back the body if no intervention, before sending upstream

Reply:
Request Body buffering implemented with https://github.com/corazawaf/coraza-proxy-wasm/pull/69